tests/test_siglent_ssa3000x_spectrum_analyzer_data_client.py

```diff
@@ -0,0 +1,88 @@
+# This file is part of ts_ess_csc.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import collections
+import contextlib
+import logging
+import types
+import unittest
+
+import numpy as np
+from lsst.ts import salobj, utils
+from lsst.ts.ess import csc
+
+logging.basicConfig(
+    format="%(asctime)s:%(levelname)s:%(name)s:%(message)s", level=logging.DEBUG
+)
+
+
+class DataClientTestCase(unittest.IsolatedAsyncioTestCase):
+    def setUp(self) -> None:
+        salobj.set_random_lsst_dds_partition_prefix()
+        config_schema = (
+            csc.SiglentSsa3000xSpectrumAnalyzerDataClient.get_config_schema()
+        )
+        validator = salobj.DefaultingValidator(config_schema)
+        config_dict = dict(
+            host="localhost",
+            port=5000,
+            connect_timeout=5,
+            read_timeout=1,
+            location="Test Location",
+            sensor_name="MockSSA3000X",
+            sleep_interval=0.1,
+            start_frequency="0.0 GHz",
+            stop_frequency="3.0 GHz",
+        )
+        config_dict = validator.validate(config_dict)
+        self.config = types.SimpleNamespace(**config_dict)
+
+    @contextlib.asynccontextmanager
+    async def create_controller(self) -> collections.abc.AsyncGenerator:
+        """Create an ESS Controller and Remote and wait for them to start."""
+        index = next(utils.index_generator())
```

Comment:
This will always set index to 1 because it always calls `next` on a new index generator.

I suggest making the index generator at the global level, by calling `utils.index_generator()`, and then call next on that. Another option is to do it in the setUpClass method. (Doing it in setUp will not be very useful, as each test will still start with index=1).

All that said, using a fixed index is also fine, instead of cycling through indices.

(If you see this issue in any of the other ts_ess_csc unit tests, please fix it there, as well).

Reply:
I modified it by calling `utils.index_generator()` and I hope I did it correctly.

The only other unit test that creates a CSC (apart form the Young weather station test) is the one for the CSC itself. It uses the salobj make_csc method and that doesn't pass on an index at all. Not sure if we should change that.

Reply:
Unfortunately not. See a comment I just added.